docs/howto/configure/auth-management.md

```diff
@@ -364,48 +364,115 @@ The steps to enable the JupyterHub CILogonOAuthenticator for a hub are simmilar
+        ```
+
+    2. **Authenticate using Google with CILogon, allowing only a certain domain**:
+
+        *This example sets `2i2c.org` as the only domain that can login into the hub using Google*
+
+        ```yaml
+        jupyterhub:
+          hub:
+            config:
+              JupyterHub:
+                authenticator_class: cilogon
+              CILogonOAuthenticator:
+                oauth_callback_url: https://{{ HUB_DOMAIN }}/hub/oauth_callback
+                allowed_idps:
+                  http://google.com/accounts/o8/id:
+                    username_derivation:
+                      username_claim: "email"
+                    allowed_domains:
+                        - "2i2c.org"
+        ```
+
+    3. **Authenticate using an instutional identity provider for the hub community users and Google for 2i2c staff.**
+
+        This example:
+          - only shows the ANU identity provider and Google as the possible login option though CILogon
+          - sets `2i2c.org` as the only domain that can login into the hub using Google
+          - sets `anu.edu.au` as the only domain that can login into the hub using the ANU institutional id provider
+          - adds a `2i2c:` prefix to the usernames logging in through Google. The hub usernames are the `email` addresses of these accounts, as specified through `username_claim`.
+          - strips the `@anu.edu.au` domain from the usernames logging in through ANU IDP. The hub usernames are the `email` addresses of these accounts, as specified through `username_claim`.
+
+      ```{note}
+      To get the value of the key that must go in the `allowed_idp` dict for a specific IdP, go to https://cilogon.org/idplist and get the value of the `EntityID` key of the desired institutional IdP.
+      ```
+
+      Example config:
+
+      ```yaml
+      jupyterhub:
+        hub:
+          config:
+            JupyterHub:
+              authenticator_class: cilogon
+            CILogonOAuthenticator:
+              oauth_callback_url: https://{{ HUB_DOMAIN }}/hub/oauth_callback
+              admin_users:
+                - admin@anu.edu.au
+              # Only show the option to login with Google and ANU
+              shown_idps:
+                - https://idp2.anu.edu.au/idp/shibboleth
+                - https://accounts.google.com/o/oauth2/auth
+              allowed_idps:
+                http://google.com/accounts/o8/id:
+                  username_derivation:
+                    username_claim: "email"
+                    action: "prefix"
+                    prefix: "2i2c"
+                  allowed_domains:
+                    - "2i2c.org"
+                https://idp2.anu.edu.au/idp/shibboleth:
+                  username_derivation:
+                    username_claim: "email"
+                    action: "strip_idp_domain",
+                    domain: "anu.edu.au"
+                  allowed_domains:
+                    - "anu.edu.au"
+      ```
```

Comment:
```suggestion
          - only shows Shibboleth, the ANU identity provider, and Google as the possible login options though CILogon
          - sets `2i2c.org` as the only domain that can login to the hub using Google
          - sets `anu.edu.au` as the only domain that can login to the hub using Shibboleth
          - adds a `2i2c:` prefix to the usernames logging in through Google. The hub usernames are the `email` addresses of these accounts, as specified through `username_claim`.
          - strips the `@anu.edu.au` domain from the usernames logging in through Shibboleth. The hub usernames are the `email` addresses of these accounts, as specified through `username_claim`.

      ```{note}
      To get the value of the key that must go in the `allowed_idp` dict for a specific IdP, go to https://cilogon.org/idplist and get the value of the `EntityID` key of the desired institutional IdP.
      ```

      Example config:

      ```yaml
      jupyterhub:
        hub:
          config:
            JupyterHub:
              authenticator_class: cilogon
            CILogonOAuthenticator:
              oauth_callback_url: https://{{ HUB_DOMAIN }}/hub/oauth_callback
              admin_users:
                - admin@anu.edu.au
              # Only show the option to login with Google and Shibboleth
              shown_idps:
                - https://idp2.anu.edu.au/idp/shibboleth
                - https://accounts.google.com/o/oauth2/auth
              allowed_idps:
                http://google.com/accounts/o8/id:
                  username_derivation:
                    username_claim: "email"
                    action: "prefix"
                    prefix: "2i2c"
                  allowed_domains:
                    - "2i2c.org"
                https://idp2.anu.edu.au/idp/shibboleth:
                  username_derivation:
                    username_claim: "email"
                    action: "strip_idp_domain",
                    domain: "anu.edu.au"
                  allowed_domains:
                    - "anu.edu.au"
      ```
```

I changed the wording slightly as it sounded like ANU were hosting their own identity provider, when I think they are actually subscribed to Shibboleth (as many universities are)

Reply:
> I changed the wording slightly as it sounded like ANU were hosting their own identity provider, when I think they are actually subscribed to Shibboleth (as many universities are)

I believe that's ok. Whatever makes more sense for everybody. After reading the CILogon docs and refactoring the oauthenticators, I admit I am confused about the terminology they use and I realized there isn't necessarily a convention on identity provider vs entity id, the use of username claims vs username keys 🤯 etc
So thank you ✨